gnovm/pkg/gnolang/values.go

```diff
@@ -2312,12 +2316,8 @@ func (b *Block) GetPointerTo(store Store, path ValuePath) PointerValue {
 	// the generation for uverse is 0.  If path.Depth is
 	// 0, it implies that b == uverse, and the condition
 	// would fail as if it were 1.
-	i := uint8(1)
-LOOP:
-	if i < path.Depth {
+	for i := uint8(1); i < path.Depth; i++ {
```

Comment:
we should probably revert, from what i remember this function can be inlined without a for-loop. For loops cannot be inlined. (we should add a comment about inlines though, so there are no modifications without measuring inlineability.

Reply:
The function cannot be inlined even with `goto`. On mater:

```
// cd path/to/pkg/gnolang
go build -gcflags=-m=3 2>&1 | less
// search for: Block).GetPointerTo[^I]
```

```
./values.go:2296:6: cannot inline (*Block).GetPointerTo: function too complex: cost 270 exceeds budget 80
```

so off-budget, and not even by a small amount :)

Reply:
I think because debug=true for you.

./values.go:2296:6: can inline (*Block).GetPointerToInt with cost 78 as: method(*Block) func(Store, int) PointerValue { vv := fillValueTV(store, &b.Values[index]); return PointerValue{...} }

Reply:
My mistake was maybe that I didn't do enough commenting while coding. But any time the code has this style of looping it's prob because of inlining.

Reply:
A-ha! You fell into the same trap I did, and I tried to guard you against!

You see, you put the log for GetPointer**ToInt**, not GetPointerTo :) Search with `Block).GetPointerTo[^I]` (note the final `[^I]` to get the right function.

Using the search regex I pointed out, after changing debug.go to have `debug` be a constant and thus optimizing out the `if debug` above statement, the result is still "too expensive" with the if/goto:

```
./values.go:2321:6: cannot inline (*Block).GetPointerTo: function too complex: cost 194 exceeds budget 80
```